src/engine/controls/cuecontrol.cpp

```diff
@@ -1861,12 +1878,20 @@ void HotcueControl::setCue(CuePointer pCue) {
     // because we have a null check for valid data else where in the code
     m_pCue = pCue;
 }
-PredefinedColorPointer HotcueControl::getColor() const {
-    return Color::kPredefinedColorsSet.predefinedColorFromId(m_hotcueColor->get());
+mixxx::RgbColor::optional_t HotcueControl::getColor() const {
+    double value = m_hotcueColor->get();
+    if (value < 0) {
+        return std::nullopt;
+    }
+    return mixxx::RgbColor(value);
```

Comment:
Maybe here also, i.e. range check and type cast. All those incoming/outgoing double conversions should be validated.

Reply:
I added a static cast. We already check if `value < 0`, or you want to check max values as well?

Reply:
If this is an input value that may be set from external components/scripts the range needs to be checked, too.

Reply:
I would expect it to be truncated to 32 bits. Isn't that the case anyway?

Reply:
I guess so. Passing a double to an integer without an explicit cast feels weird. I would expect at least a compiler warning.

We don't need a range check since RgbColor clamps the value. Maybe we could add a debug assertion to detect rounding issues or out-of-range values?

```
DEBUG_ASSERT(value == mixxx::RgbColor::validateCode(static_cast<mixxx::RgbColor::code_t>(value)));
```